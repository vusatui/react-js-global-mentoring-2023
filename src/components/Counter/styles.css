.v-counter > span {
    padding: 0 15px;
}

.v-counter > button {
    cursor: pointer;
}
