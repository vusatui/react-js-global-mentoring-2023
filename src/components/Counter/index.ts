import React, {CSSProperties} from "react";

import "./styles.css";

interface Props {
    style?: CSSProperties
    step?: number
    value: number
    onChange: (v: number) => void
}

const Counter = ({
    style = undefined,
    step = 1,
    value,
    onChange,
}: Props) => {
    const increment = () => onChange(value + step);

    const decrement = () => onChange(value - step);

    return React.createElement("div", { style, className: "v-counter" }, [
        React.createElement("button", { onClick: decrement }, "-"),
        React.createElement("span", {}, value),
        React.createElement("button", { onClick: increment }, "+"),
    ]);
};

export default Counter;