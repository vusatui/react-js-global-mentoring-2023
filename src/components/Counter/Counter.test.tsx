import Counter from "./index";
import {act, render, screen} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import {useState} from "react";

const INITIAL_STATE = 0;

const Component = () => {
    const [value, setValue] = useState<number>(INITIAL_STATE)
    const handleChange = (v: number) => act(() => setValue(v));
    return (<Counter value={value} onChange={handleChange} />);
};

describe("Counter", () => {

    test("Should renders correctly", () => {
        const { asFragment } = render(<Component />);
        expect(screen.getByText("0")).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    });

    test("Should decrement value", () => {
        render(<Component />);
        userEvent.click(screen.getByText("-"));
        expect(screen.getByText("-1")).toBeInTheDocument();
    });

    test("Should increment value", async () => {
        render(<Component />);
        userEvent.click(screen.getByText("+"));
        expect(screen.getByText("1")).toBeInTheDocument();
    });

});