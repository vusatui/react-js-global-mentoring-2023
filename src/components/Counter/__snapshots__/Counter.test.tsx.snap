// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Should renders correctly 1`] = `
<DocumentFragment>
  <div
    class="v-counter"
  >
    <button>
      -
    </button>
    <span>
      0
    </span>
    <button>
      +
    </button>
  </div>
</DocumentFragment>
`;
